homeassistant/components/drop_connect/config_flow.py

```diff
@@ -0,0 +1,153 @@
+from homeassistant.util.json import JSON_DECODE_EXCEPTIONS, json_loads
+
+from .const import (
+    CONF_COMMAND_TOPIC,
+    CONF_DATA_TOPIC,
+    CONF_DEVICE_DESC,
+    CONF_DEVICE_ID,
+    CONF_DEVICE_NAME,
+    CONF_DEVICE_TYPE,
+    CONF_HUB_ID,
+    CONF_UNIQUE_ID,
+    DISCOVERY_TOPIC,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+KEY_DEVICE_TYPE = "devType"
+KEY_DEVICE_DESCRIPTION = "devDesc"
+KEY_DEVICE_NAME = "name"
+
+
+class FlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle DROP config flow."""
+
+    VERSION = 1
+
+    _discovered_device_info: dict[str, Any] = {}
+    _topic = DISCOVERY_TOPIC
+    __command_topic: str | None = None
+    __data_topic: str | None = None
+    __device_desc: str | None = None
+    __device_id: str | None = None
+    __device_type: str | None = None
+    __hub_id: str | None = None
+    __name: str = ""
+    __unique_id: str | None = None
+
+    async def async_step_mqtt(self, discovery_info: MqttServiceInfo) -> FlowResult:
+        """Handle a flow initialized by MQTT discovery."""
+
+        # Abort if the topic does not match our discovery topic or the payload is empty.
+        if (
+            discovery_info.subscribed_topic != DISCOVERY_TOPIC
+            or not discovery_info.payload
+        ):
+            return self.async_abort(reason="invalid_discovery_info")
+
+        payloadData = {}
+        try:
+            json_dict = (
+                json_loads(discovery_info.payload)
+                if isinstance(discovery_info.payload, str)
+                else None
+            )
+            if isinstance(json_dict, dict):
+                for k, v in json_dict.items():
+                    if isinstance(k, str) and isinstance(v, str):
+                        payloadData[k] = v
+        except JSON_DECODE_EXCEPTIONS:
+            _LOGGER.error(
+                "Invalid MQTT discovery payload on %s: %s",
+                discovery_info.topic,
+                discovery_info.payload,
+            )
+            return self.async_abort(reason="invalid_discovery_info")
+
+        # Extract the DROP hub ID and DROP device ID from the MQTT topic.
+        topicElements = discovery_info.topic.split("/")
+        if not (topicElements[2].startswith("DROP-") and topicElements[3].isnumeric()):
+            return self.async_abort(reason="invalid_discovery_info")
+        self.__hub_id = topicElements[2]
+        self.__device_id = topicElements[3]
+
+        # Build the unique ID and register it.
+        self.__unique_id = f"{self.__hub_id}_{self.__device_id}"
+        await self.async_set_unique_id(self.__unique_id)
+
+        # Abort if this device has already been configured.
+        already_configured = self._async_current_ids(False)
+        if self.__unique_id in already_configured:
+            _LOGGER.debug("Device %s already configured", self.__unique_id)
+            return self.async_abort(reason="invalid_discovery_info")
```

Comment:
I simplified this code in 8f5a4a8d0eca588b7ca12148aae7f27c5bedbf54 and implemented part of your suggestions, but the use of `_abort_if_unique_id_configured()` has negative side-effects.  By replacing the check if the unique_id is in `_async_current_ids()' with `_abort_if_unique_id_configured()`, I end up with duplicate devices showing up in the HA UI when the DROP Hub broadcasts its discovery data.  I'm going to mark this as resolved because it behaves as expected, but feel free to reopen this issue if you still have doubts after reviewing the latest changes.